GetAmbush/ambush-app
# Issue 3: Implement JSON Import and Export Backup Functionality - Closes #3

# Description:
Thir PR implements a feature that allows exporting data from the application, which is saved in the browser preferences to a JSON file as a back up and importing again to recover the back up in case user loses data across eventual migrations or data cleaning(Safari for example).

Closes #3

# Implemented features

1. Export JSON back up to computer file system
2. Import JSON back up from file system
3. Format error handling
4. UI to recover back up from onboarding root screen and invoice list page, and download back up from invoice list page.

# How to test
1. Run the application 
2. Follow through the onboarding filling all required information
3. On invoice list page, after creating some new invoices, tap the download icon on top bar to download all data in a JSON file
4. Check JSON file is valid
5. Run application in another browser/port and at onboarding screen, click to `Retrieve from back up` and select the saved JSON file.
6. Check that all invoices are rendered to the list and verify all settings were correctly uploaded in their respective sections.
7. Change any configuration in the JSON file and from the invoice list page, upload the file again to check it was correctly saved.

# Edge cases
1. In onboarding screen, write a JSON file in a wrong format and try to upload to the application to receive an error dialog
2. Repeat previous step in invoice list page

# Evidence

https://github.com/user-attachments/assets/b1afa88e-eb7e-4b06-b495-45acd6033dc9
